crates/amaru-kernel/src/serde_utils.rs

```diff
@@ -107,3 +110,28 @@ impl From<ScriptRefProxy> for ScriptRef {
         }
     }
 }
+
+// ------------------------------------------------------------------------------- DatumOption
+
+#[derive(Debug, serde::Deserialize)]
+pub enum DatumOptionProxy {
+    Hash(Bytes),
+    Data(Bytes),
+}
+
+impl HasProxy for DatumOption {
+    type Proxy = DatumOptionProxy;
+}
+
+impl From<DatumOptionProxy> for DatumOption {
+    #[allow(clippy::unwrap_used)]
+    fn from(value: DatumOptionProxy) -> Self {
+        match value {
+            DatumOptionProxy::Hash(bytes) => DatumOption::Hash(Hash::from(bytes.as_slice())),
+            // This code should only be run during tests, so a panic here is fine
+            DatumOptionProxy::Data(data) => {
+                DatumOption::Data(CborWrap(from_cbor(data.deref()).unwrap()))
+            }
+        }
+    }
+}
```

Comment:
Perhaps, keep this function safe by defining `TryFrom` and move `unwrap` at call-site where the context makes it reasonable (or not) to be used (and same for `ScriptRefProxy` actually).

Reply:
The `HasProxy` logic requires a `From`, which is why I went with this solution. I don't love the moving of the `unwrap` to the call-site, as it only moves it one level up to the `OutputProxy`, which doesn't handle a result. We could refactor the whole thing, which feels more involved, or we could leave it as is as this code is only used in test and it doesn't *really* matter if this function is safe. Happy to do either one, *more* happy to leave it as is though 😛

Reply:
If it's only for tests we might want to hide it behind a `#[cfg(test)]` annotation.